Addons/properties-templates/naviox.properties

```diff
@@ -6,6 +6,9 @@ storePasswordAsHex=true
 initialPasswordForAdmin=admin
 # There are many more password options in Configuration module of XavaPro (http://www.openxava.org/xavapro)
 
+# Application Name
+applicationName=demo

```

Comment:
Ideally adding app name in more places should be avoided.

@javierpaniza what are your thoughts on this being read from application.xml itself ?  

I wanted to ask you separately whether this application name is a legacy thing or still has some value in the future

Reply:
Application name in the URL should be the standard way, and deploying on the root context a specific usage, because having several OpenXava applications in the same server is a typical pattern, for example in openxava.org we use the ROOT context as the main site, and we have deployed on it several OpenXava applications that are acceded from the site, like the demos or the issue tracker.

Reply:
@javierpaniza from a web application point of view the kosher approach (Particularly to Tomcat) is to name the WAR file the context desired. For example, Invoice Demo WAR should be named InvoiceDemo.WAR if non root context path is desired.  Alternatively naming it ROOT.WAR if root context is desired. Tomcat will do the rest.

Referring to existence of (or non existence) of context path in code should be avoided. 
OpenXava prior to this patch expects a context path and it doesn't work when WAR is named ROOT.war.

Hope this helped to clear the intention here. 

1. My particular code comment was about how to fetch the application name now that we should not assume it can come from context path. @HarshaMahaswami has done this by putting another property in naviox.properties. I was asking you if we can just directly read from application.xml itself
2. Deeper question is the relevance of this value in application.xml itself as for each web app there will only be one entry. This does appear to be legacy.

Reply:
> I was asking you if we can just directly read from application.xml itself

Yes, you can. Indeed I refined your code in order to do so. The classes to read application.xml are MetaApplication and MetaApplications. I just added an getMainMetaApplication() to MetaApplications.

>  application.xml itself as for each web app there will only be one entry

We still need an application name. For example, when a portal page (like Liferay) we have portlets from several applications in the same page, it works nicely, because all ids in the HTML used the application name in the id. This is only an example, the application name is widely used, we cannot remove it.

Reply:
Understood your use cases for application.xml.